godtools/App/Share/Data/MobileContentAuthTokenRepository/MobileContentAuthTokenRepository.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  MobileContentAuthTokenRepository.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 10/31/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+import OktaAuthentication
+import Combine
+
+class MobileContentAuthTokenRepository {
+    
+    private let api: MobileContentAuthTokenAPI
+    private let cache: MobileContentAuthTokenCache
+    
+    private var cancellables = Set<AnyCancellable>()
+    
+    init(api: MobileContentAuthTokenAPI, cache: MobileContentAuthTokenCache) {
+        
+        self.api = api
+        self.cache = cache
+    }
+    
+    func getAuthTokenPublisher(for userId: Int? = nil, oktaAccessToken: String) -> AnyPublisher<String?, URLResponseError> {
+        
+        if let authToken = getCachedAuthToken(for: userId) {
+            
+            return Just(authToken)
+                .setFailureType(to: URLResponseError.self)
+                .eraseToAnyPublisher()
+            
+        } else {
+            
+            return fetchAuthToken(oktaAccessToken: oktaAccessToken)
+        }
+    }
+
+    private func getCachedAuthToken(for userId: Int?) -> String? {
+        
+        guard let userId = userId else { return nil }
+        
+        return cache.getAuthToken(for: userId)
+    }
+    
+    private func fetchAuthToken(oktaAccessToken: String) -> AnyPublisher<String?, URLResponseError> {
```

Comment:
Hey @rachaelblue I think this method would be nice to have exposed.  The ```AuthenticateUserUseCase``` could then make use of this and fetch a new auth token when a user authenticates.
It would also be nice to include in the method name that this is fetching a token from Remote too.